File: common/src/client/java/juuxel/adorn/client/gui/screen/DrawerScreen.java
package juuxel.adorn.client.gui.screen;

import juuxel.adorn.AdornCommon;
import juuxel.adorn.menu.DrawerMenu;
import net.minecraft.network.chat.Component;
import net.minecraft.resources.Identifier;
import net.minecraft.world.entity.player.Inventory;

public final class DrawerScreen extends PalettedMenuScreen<DrawerMenu> {
    private static final Identifier BACKGROUND_TEXTURE = AdornCommon.id("textures/gui/drawer.png");
    private static final Identifier PALETTE_ID = AdornCommon.id("drawer");

    public DrawerScreen(DrawerMenu menu, Inventory playerInventory, Component title) {
        super(menu, playerInventory, title);
    }

    @Override
    protected Identifier getBackgroundTexture() {
        return BACKGROUND_TEXTURE;
    }

    @Override
    protected Identifier getPaletteId() {
        return PALETTE_ID;
    }
}

File: common/src/main/kotlin/juuxel/adorn/client/gui/screen/DrawerScreen.kt
package juuxel.adorn.client.gui.screen

import juuxel.adorn.AdornCommon
import juuxel.adorn.menu.DrawerMenu
import net.minecraft.entity.player.PlayerInventory
import net.minecraft.text.Text

class DrawerScreen(menu: DrawerMenu, playerInventory: PlayerInventory, title: Text) : PalettedMenuScreen<DrawerMenu>(menu, playerInventory, title) {
    override val backgroundTexture = BACKGROUND_TEXTURE
    override val paletteId = PALETTE_ID

    companion object {
        private val BACKGROUND_TEXTURE = AdornCommon.id("textures/gui/drawer.png")
        private val PALETTE_ID = AdornCommon.id("drawer")
    }
}
